### PR TITLE
Ani-2x dataloader

### PR DESCRIPTION
**Pull request by stefdoerr**
- Added a data loader for ANI-2x
- Fixed a bug with variable names for element energies (can you confirm this was wrong?)
- Fixed a bug where training would crash if log folder was missing

### CODE REVIEW

File: torchmdnet/datasets/ani.py [hunk 1]
```diff
@@ -206,12 +208,12 @@ def get(self, idx):
 class ANI1(ANIBase):
     __doc__ = ANIBase.__doc__
     # Avoid sphinx from documenting this
-    ELEMENT_ENERGIES = {
```
Comment (stefdoerr) [inline torchmdnet/datasets/ani.py:209]: In line 49 there is this code `energy = sum(self._ELEMENT_ENERGIES[z] for z in atomic_numbers)`
which seems to assume that the classes have a `_ELEMENT_ENERGIES` variable. However here it's missing the underscore. Am I right in changing this?

File: torchmdnet/scripts/train.py [hunk 2]
```diff
@@ -130,6 +131,7 @@ def get_args():
     if args.inference_batch_size is None:
         args.inference_batch_size = args.batch_size
 
+    os.makedirs(os.path.abspath(args.log_dir), exist_ok=True)
```
Comment (stefdoerr) [inline torchmdnet/scripts/train.py:134]: Creates the log folder instead of getting an error in save_argparse

Comment (RaulPPelaez) [reply torchmdnet/scripts/train.py:134]: Damn thanks for this. I was really tired of having to create the folder.

File: torchmdnet/datasets/ani.py (hunk 1 above)
Comment (RaulPPelaez) [reply torchmdnet/datasets/ani.py:209]: Definetely a bug. Thanks!

**Review by RaulPPelaez — APPROVED** [review]

**Comment by RaulPPelaez** [comment]
Thanks @stefdoerr ! ready to merge?

File: torchmdnet/scripts/train.py (hunk 2 above)
Comment (guillemsimeon) [reply torchmdnet/scripts/train.py:134]: 100%

**Comment by stefdoerr** [comment]
Yeah fine by me

**Pull request merged**